module/input/SensorFilter/src/SensorFilter.cpp

```diff
@@ -564,125 +571,133 @@ namespace module::input {
                                 const auto filterState = MotionModel<double>::StateVec(motionFilter.get());
                                 Eigen::Affine3d Hwt;
                                 Hwt.linear()      = filterState.Rwt.toRotationMatrix();
                                 Hwt.translation() = filterState.rTWw;
-
+                                // Htg is intended to be such that the "foot down" position is where the foot would be
+                                // if it were flat, even if it's not flat when first touches the ground. As the foot
+                                // flattens, it's meant to becomes true. This means that even if the foot hits the
+                                // ground at an angle, it doesn't store that angled position as the footlanding_Hwf, but
+                                // instead stores the position that foot would be if/when it becomes flat on the ground
                                 Eigen::Affine3d Htg(utility::motion::kinematics::calculateGroundSpace(Htf, Hwt));
 
                                 footlanding_Hwf[side]                   = Hwt * Htg;
                                 footlanding_Hwf[side].translation().z() = 0.0;
 
+                                // Store the current foot down state for next time
                                 previous_foot_down[side] = true;
                             }
+                            // Else is down, and didn't hit the ground this time step
                             else if (foot_down && prev_foot_down) {
                                 // Use stored Hwf and Htf to calculate Hwt
-                                Eigen::Affine3d footlanding_Hwt = footlanding_Hwf[side] * Htf.inverse();
+                                const Eigen::Affine3d footlanding_Hwt = footlanding_Hwf[side] * Htf.inverse();
 
                                 // do a foot based position update
                                 motionFilter.measure(Eigen::Vector3d(footlanding_Hwt.translation()),
                                                      config.motionFilter.noise.measurement.flatFootOdometry,
                                                      MeasurementType::FLAT_FOOT_ODOMETRY());
 
                                 // do a foot based orientation update
-                                Eigen::Quaterniond Rwt(footlanding_Hwt.linear());
+                                const Eigen::Quaterniond Rwt(footlanding_Hwt.linear());
                                 motionFilter.measure(Rwt.coeffs(),
                                                      config.motionFilter.noise.measurement.flatFootOrientation,
                                                      MeasurementType::FLAT_FOOT_ORIENTATION());
                             }
+                            // Else the foot is off the ground, so we make sure that for the next time step, we know
+                            // that this time step, the foot was off the ground
```

Comment:
```suggestion
                            // Otherwise this sides foot is off the ground, so we make sure that for the next time step, we know
                            // that this time step, the foot was off the ground
```